test/unittests/test_dd_synthesis.cpp

```diff
@@ -0,0 +1,248 @@
+#include "algorithms/synthesis/dd_synthesis.hpp"
+#include "algorithms/synthesis/encoding.hpp"
+#include "core/io/pla_parser.hpp"
+#include "dd/FunctionalityConstruction.hpp"
+
+#include "gtest/gtest.h"
+
+using namespace dd::literals;
+using namespace syrec;
+
+class TestDDSynth: public testing::Test {
+protected:
+    TruthTable                     tt{};
+    std::string                    testCircuitsDir = "./circuits/";
+    std::unique_ptr<dd::Package<>> ddSynth         = std::make_unique<dd::Package<>>(6U);
+};
+
+TEST_F(TestDDSynth, CUSTOMTT1) {
+    std::string circCUSTOMTT1 = testCircuitsDir + "CUSTOMTT1.pla";
+
+    EXPECT_TRUE(readPla(tt, circCUSTOMTT1));
+
+    EXPECT_EQ(tt.size(), 7U);
+
+    extend(tt);
+
+    EXPECT_EQ(tt.size(), 8U);
+
+    EXPECT_EQ(tt.nInputs(), 3U);
+    EXPECT_EQ(tt.nOutputs(), 3U);
+
+    auto ttDD = buildDD(tt, ddSynth);
+
+    EXPECT_TRUE(ttDD.p != nullptr);
+
+    EXPECT_NE(ttDD, ddSynth->makeIdent(3U));
+
+    DDSynthesis quantumCircuit(3U);
+
+    auto trueMatrix  = ddSynth->getMatrix(ttDD);
+    auto synthMatrix = ddSynth->getMatrix(dd::buildFunctionality(&quantumCircuit.synthesize(ttDD, ddSynth), ddSynth));
+
+    EXPECT_EQ(trueMatrix, synthMatrix);
+
+    EXPECT_EQ(ttDD, ddSynth->makeIdent(3U));
+}
+
+TEST_F(TestDDSynth, CUSTOMTT2) {
+    std::string circCUSTOMTT2 = testCircuitsDir + "CUSTOMTT2.pla";
+
+    EXPECT_TRUE(readPla(tt, circCUSTOMTT2));
+
+    EXPECT_EQ(tt.size(), 7U);
+
+    extend(tt);
+
+    EXPECT_EQ(tt.size(), 8U);
+
+    EXPECT_EQ(tt.nInputs(), 3U);
+    EXPECT_EQ(tt.nOutputs(), 3U);
+
+    auto ttDD = buildDD(tt, ddSynth);
+    EXPECT_TRUE(ttDD.p != nullptr);
+
+    EXPECT_NE(ttDD, ddSynth->makeIdent(3U));
+
+    DDSynthesis quantumCircuit(3U);
+
+    auto trueMatrix  = ddSynth->getMatrix(ttDD);
+    auto synthMatrix = ddSynth->getMatrix(dd::buildFunctionality(&quantumCircuit.synthesize(ttDD, ddSynth), ddSynth));
+
+    EXPECT_EQ(trueMatrix, synthMatrix);
+
+    EXPECT_EQ(ttDD, ddSynth->makeIdent(3U));
+}
```

Comment:
This test only differs from the previous test by the filename. Reduce code duplication by making this a parametrized test.

Actually, this more or less holds for all the other tests as well. All of them are defined by a file name and a qubit number. Correspondingly unify the tests.

Reply:
done 👍🏻